docs/old-reference-guide/modules/axon-framework-commands/pages/command-handlers.adoc

```diff
@@ -1,249 +1,928 @@
 
-To define a Command Handler in an Aggregate, simply annotate the method which should handle the command with `@CommandHandler`.
-The `@CommandHandler` annotated method will become a Command Handler for Command Messages where the _command name_ matches fully qualified class name of the first parameter of that method.
-Thus, a method signature of `void handle(RedeemCardCommand cmd)` annotated with `@CommandHandler`, will be the Command Handler of the `RedeemCardCommand` Command Messages.
+== Stateless vs stateful command handlers
 
-Command Messages can also be xref:command-dispatchers.adoc[dispatched] with different _command names_.
-To be able to handle those correctly, the `String commandName` value can be specified in the `@CommandHandler` annotation.
+Command handlers in Axon 5 fall into two categories:
 
-In order for Axon to know which instance of an Aggregate type should handle the Command Message, the property carrying the Aggregate Identifier in the command object *must* be annotated with `@TargetAggregateIdentifier`.
-The annotation may be placed on either the field or an accessor method (for example, a getter) in the Command object.
+* **Stateless command handlers**: Regular components (Spring beans or registered objects) that don't maintain state in the handler itself between command invocations.
+Each command is processed independently.
+State may be accessed through injected services or repositories, but the handler itself doesn't manage state.
+
+* **Stateful command handlers**: Components that maintain state in the handler itself across command invocations.
+These are typically event-sourced entities where state is built up from events and stored in the handler instance.
 
 [NOTE]
-.Routing in a distributed environment
 ====
-Regardless of the type of command, as soon as you start distributing your application (through Axon Server, for example), it is recommended to specify a routing key on the command.
-This is the job of the `@TargetAggregateIdentifier`, but in absence of a field worthy of the annotation, the `@RoutingKey` annotation should be added to ensure the command can be routed.
+In Axon 5, the emphasis has shifted from modeling-centric (Axon 4's aggregate focus) to command-centric.
+Entities are now understood as one implementation pattern for stateful command handling, not a core framework concept.
+The focus is on handling commands effectively, with various patterns available to achieve that goal.
+====
+
+== Stateless command handlers
+
+Stateless command handlers are regular components with `@CommandHandler` methods that don't maintain state in the handler itself.
+They may access state through injected services or repositories, but the handler instance itself is stateless.
+
+[source,java]
+----
+import org.axonframework.messaging.commandhandling.CommandHandler;
```

Comment:
the `CommandHandler` annotation resides in `org.axonframework.messaging.commandhandling.annotation.CommandHandler` after the package refactoring